src/main/Extensions/ApplicationSearch/Windows/WindowsApplicationRepository.ts

```diff
@@ -30,7 +32,14 @@ export class WindowsApplicationRepository implements ApplicationRepository {
     }
 
     private async getManuallyInstalledApps(): Promise<WindowsApplication[]> {
-        const folderPaths = this.settings.getValue<string[]>("windowsFolders");
+        const folderPaths: string[] = [];
+        this.settings.getValue<string[]>("windowsFolders").forEach((folderPath) => {
+            if (this.fileSystemUtility.isDirectory(folderPath)) {
+                folderPaths.push(folderPath);
+            } else {
+                this.logger.warn(`${folderPath} does not exist or is not a folder.`);
+            }
+        });
```

Comment:
Thanks for your contribution. Generally I'm happy with this change, but I would like to have a few adjustments:

1. Please extract the filtering of existing folder paths to a dedicated method
2. I would like to replace `.forEach()` with `for (const item of items)` for the following reasons:
  - Staying consistent with the rest of the codebase, we almost exclusively use `for .. of` loops (I know that the `LinuxApplicationRepository` also has a `.forEach` loop, I'll adjust this in a follow-up)
  - `for .. of `loops support `await` inside the loop
  - Easier to break, continue, or return
  - Slightly better performance in some cases
  - More familiar if coming from other programming languages
3. Please add a test that covers the issue that your change is fixing

Reply:
Should be good